plugins/node/opentelemetry-instrumentation-mongodb/src/mongodb.ts

```diff
@@ -0,0 +1,414 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+  getSpan,
+  SpanStatusCode,
+  Span,
+  context,
+  diag,
+  SpanKind,
+} from '@opentelemetry/api';
+import type * as mongodb from 'mongodb';
+import {
+  MongodbCommandType,
+  MongoInternalCommand,
+  MongoInternalTopology,
+  WireProtocolInternal,
+  MongoDbInstrumentationConfig,
+  CursorState,
+} from './types';
+import { VERSION } from './version';
+import {
+  DatabaseAttribute,
+  GeneralAttribute,
+} from '@opentelemetry/semantic-conventions';
+import {
+  InstrumentationBase,
+  InstrumentationNodeModuleDefinition,
+  InstrumentationNodeModuleFile,
+  isWrapped,
+} from '@opentelemetry/instrumentation';
+
+/** mongodb instrumentation plugin for OpenTelemetry */
+export class MongoDBInstrumentation extends InstrumentationBase<
+  typeof mongodb
+> {
+  constructor(protected _config: MongoDbInstrumentationConfig = {}) {
+    super('@opentelemetry/instrumentation-mongodb', VERSION, _config);
+  }
+
+  init() {
+    const { patch, unpatch } = this._getPatches();
+    return [
+      new InstrumentationNodeModuleDefinition<typeof mongodb>(
+        'mongodb',
+        this._supportedVersions,
+        undefined,
+        undefined,
+        [
+          new InstrumentationNodeModuleFile<WireProtocolInternal>(
+            'mongodb/lib/core/wireprotocol/index.js',
```

Comment:
This doesn't work on windows because of the `/`. Using `\\` works (but most likely breaks non windows).

Reply:
Have you tested this on windows? I think this uses the node module resolver which I thought normalized paths.

Reply:
Even though i think its handled by node, if thats an actual bug we should fix this directly in the instrumentation base package because the mongodb plugin isn't the only one to use internals require

Reply:
Yes, tested on windows.

Reply:
@Flarna Could you open an issue on the main repo with the details ? we should fix this before RC of SDK for sure

Reply:
https://github.com/open-telemetry/opentelemetry-js/issues/1990

Reply:
What about using `path.normalize('mongodb/lib/core/wireprotocol/index.js'),` for now? The turnaround time for a fix in core until it arrives in contrib is not that short.

Reply:
I'm fine doing that in another PR so we can do one PR for all contrib, WDYT ?

Reply:
I think we should make a fix in core repo when `InstrumentationNodeModuleFile` is being created us path.normalize ass @Flarna mentioned